Update requirement of homematicip_cloud component to v1.0.1

web-dc opened the pull request:
## Proposed change

Update homematicip rest library
Update to homematicip rest api v1.0.1 to fix issues with unknown ACCESS_CONTROL and SECURITY_AND_ALARM

https://github.com/coreGreenberet/homematicip-rest-api/compare/0.13.1...1.0.1

## Type of change
Update requirement of homematicip_cloud component

- [X] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #50586
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist


- [X] The code change is tested and works locally.
- [X] Local tests pass. **Your PR cannot be merged unless tests pass**
- [X] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:


- [X] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum


To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone


[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

Link to changes from previous dependency version to new one: https://github.com/coreGreenberet/homematicip-rest-api/compare/0.13.1...1.0.1

web-dc committed:
Issue #50586
Update homematicip rest library

homeassistant commented:
Hi @web-dc, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;core#51407).

Once you do that we will be able to review and accept this pull request.

Thanks!

frenck commented:
Thanks for bumping the dependency! However, could you please update the PR description to contain at least one (or multiple) of the following:

- A link to the release notes of this package version, and all versions in between.
- A link to the changelog of this package.
- A link to a Git(Hub) diff/compare view from the current version to the bumped version.

This allows us to review upstream changes, which is needed to decide this change is working as intended and/or if we can include it in, for example, a patch release of Home Assistant.

Thanks already! 👍

oneteam2win commented:
Quick question, is this requirement being implemented?

web-dc commented:
@oneteam2win Yes, everything is up-to-date! Anybody just need to merge it

@frenck Could you take care? I added a link to the diff to the description

frenck committed:
Run script.gen_requirements_all

frenck reviewed (changes requested):
I've run the `script.update_requirements_all` (which was not been done) and rebased the PR, however, that still leaves us with some build errors.

@web-dc Could you please take a look?

schiegg committed:
Adopt homematicip_cloud test data

Pull request home-assistant/core#51407 addresses the changes done in coreGreenberet/homematicip-rest-api#410 only by updating the dependency to the newer released version of it, making it aware of the new channel indexes.
However it currently fails the integration checks [as mentioned by @frenck](https://github.com/home-assistant/core/pull/51407#pullrequestreview-693447049).
It seems that the local homeassistant test suite also has a dedicated version of the test data written [here](https://github.com/coreGreenberet/homematicip-rest-api/blob/d7c1b2c231603603b543e57eff1c0bcd6fd5e268/homematicip_demo/json_data/home.json#L8598). 
So maybe we also have to manually sync these changes.

schiegg commented:
@web-dc I added a PR to your pending PR updating the test data potentially responsible for the failed test cases. Could you have a look at it? I searched a lot on how to best add changes to a active pending PR but as I am no maintainer it seems not to be intended to do so. I hope this way you can simply accept the PR and merge my changes.

web-dc committed:
Merge pull request #1 from schiegg/patch-1

Adopt homematicip_cloud test data

homeassistant commented:
Hi @schiegg, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;core#51407).

Once you do that we will be able to review and accept this pull request.

Thanks!

web-dc commented:
@schiegg Thanks, Looks like you Need to sign the CLA. 

@frenck Could you rerun the Script?

frenck reviewed (approved):
LGTM, thanks @web-dc 👍
